### PR TITLE
fix: add detail for APM notify

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on #72 

#### List of Changes
Add NPG `POST notify` APM details

#### Motivation and Context
Those details will be send by apim policy only for APM method onboarding containing data to be add to wallet (such as maskedEmail for paypal method)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/services/WalletService.kt [hunk 1]
```diff
@@ -463,6 +467,48 @@ class WalletService(
...
+                    if (operationDetails is WalletNotificationRequestPaypalDetailsDto) {
+                        Pair(
+                            WalletStatusDto.VALIDATED,
+                            walletDetails.copy(maskedEmail = operationDetails.maskedEmail)
+                        )
+                    } else {
+                        logger.error(
+                            "No details received for PayPal wallet, cannot retrieve maskedEmail"
+                        )
+                        Pair(WalletStatusDto.ERROR, walletDetails)
```
Comment (Fernando-Granato) [inline src/main/kotlin/it/pagopa/wallet/services/WalletService.kt:500]: Isn't it better to throw an exception in this case? throw InvalidRequestException() ?

**Review by Fernando-Granato — COMMENTED** [review]
Just a little comment 🙂

File: src/main/kotlin/it/pagopa/wallet/services/WalletService.kt (hunk 1 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/wallet/services/WalletService.kt:500]: But this way wallet will remain in `VALIDATION_REQUESTED` state. Instead doing this wallet will go to ERROR for distinguish cases where wallet did not receive POST notify from those where we receive a POST notify with erroneous data.

Comment (Fernando-Granato) [reply src/main/kotlin/it/pagopa/wallet/services/WalletService.kt:500]: Ok it comes back to me even if hypothetically it's as if we are receiving a request with the wrong body right? In that case we would have responded with a 400 without changing the status of the wallet

Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/wallet/services/WalletService.kt:518]: immagine

**Review by giovanniberti — APPROVED** [review]

File: src/main/kotlin/it/pagopa/wallet/services/WalletService.kt (hunk 1 above)
Comment (pietro-tota) [reply src/main/kotlin/it/pagopa/wallet/services/WalletService.kt:518]: Fixed with https://github.com/pagopa/pagopa-wallet-service/pull/76/commits/ac62c1a3ef0617f644a3393606ab9d1dac637e10

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**